server/src/main/java/org/apache/cassandra/sidecar/handlers/livemigration/LiveMigrationFileStreamHandler.java

```diff
@@ -0,0 +1,243 @@
+ * doesn't stream the file but relies on {@link FileStreamHandler} to do so. This handler doesn't allow
+ * using "/.." in the path to access files. This handler does not serve files which are excluded in
+ * Live Migration configuration.
+ */
+public class LiveMigrationFileStreamHandler extends AbstractHandler<Void> implements AccessProtected
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LiveMigrationFileStreamHandler.class);
+
+    private final Map<Integer, List<PathMatcher>> fileExclusionsByInstanceId = new ConcurrentHashMap<>();
+    private final Map<Integer, List<PathMatcher>> dirExclusionsByInstanceId = new ConcurrentHashMap<>();
+    private final LiveMigrationConfiguration liveMigrationConfiguration;
+
+    @Inject
+    public LiveMigrationFileStreamHandler(InstanceMetadataFetcher metadataFetcher,
+                                          ExecutorPools executorPools,
+                                          CassandraInputValidator validator,
+                                          SidecarConfiguration sidecarConfiguration)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.liveMigrationConfiguration = sidecarConfiguration.liveMigrationConfiguration();
+    }
+
+    @Override
+    protected Void extractParamsOrThrow(RoutingContext context)
+    {
+        String dirType = context.pathParam(DIR_TYPE_PARAM);
+        if (null == dirType || dirType.isEmpty() || null == LiveMigrationDirType.find(dirType))
+        {
+            throw new HttpException(HttpResponseStatus.BAD_REQUEST.code(), "Invalid directory type: " + dirType);
+        }
+
+        String dirIndex = context.pathParam(DIR_INDEX_PARAM);
+        int index;
+        try
+        {
+            index = Integer.parseInt(dirIndex);
+        }
+        catch (NumberFormatException formatException)
+        {
+            throw new HttpException(HttpResponseStatus.BAD_REQUEST.code(), "Invalid directoryIndex: " + dirIndex);
+        }
+        if (index < 0)
+        {
+            throw new HttpException(HttpResponseStatus.BAD_REQUEST.code(), "Invalid directoryIndex: " + dirIndex);
+        }
+
+        // Path params are not used further, hence returning null.
+        return null;
+    }
+
+    @Override
+    protected void handleInternal(RoutingContext rc,
+                                  HttpServerRequest httpRequest,
+                                  @NotNull String host,
+                                  SocketAddress remoteAddress,
+                                  @Nullable Void request)
+    {
+        String reqPath = URLDecoder.decode(rc.request().path(), StandardCharsets.UTF_8);
+
+        if (reqPath.contains("/../") || reqPath.endsWith("/.."))
+        {
+            LOGGER.warn("Tried to access file using relative path({}). Rejecting the request.", reqPath);
+            rc.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code()).end();
+            return;
+        }
```

Comment:
In terms of validation, how is this section different from the section below? `reqPath` is same as `normalizedPath`, right?

```
        String normalizedPath = rc.normalizedPath();
        String localFile;

        try
        {
            localFile = LiveMigrationInstanceMetadataUtil.localPath(normalizedPath, instanceMeta);
        }
        catch (IllegalArgumentException e)
        {
            LOGGER.warn("Invalid path", e);
            rc.response().setStatusCode(HttpResponseStatus.NOT_FOUND.code()).end();
            return;
        }
```